src/middlewares/validateAdmin.js

```diff
@@ -0,0 +1,48 @@
+import { respondWithWarning } from '../helpers/responseHandler';
+import { findSingleUser } from '../services/inviteServices';
+import { verifyToken, formatJWTErrorMessage } from '../helpers/jwt';
+
+/**
+   * Method to generate token
+   * @param {object} req
+   * @param {object} res
+   * @param {Function} next
+   * @returns {Function} next middleware
+   */
+export const authenticateAdminToken = (req, res, next) => {
+  let token = req.headers.authorization;
+  if (token && token.startsWith('Bearer ')) {
+    token = token.slice(7, token.length);
+  }
+  if (!token) {
+    return respondWithWarning(res, 401, 'Session Expired');
+  }
+  try {
+    const { key } = verifyToken(token);
+    req.auth = key.isAdmin();
+    return next();
+  } catch (error) {
+    return respondWithWarning(res, 401, formatJWTErrorMessage(error.message));
+  }
+};
```

Comment:
This function is redundant since there is a validatedUserToken inside authenticaion.js. Kindly delete this function and use `validatedUserToken` as the first middleware in route.

This is how your route should look like
app.patch('/api/v1/users/:id', vaidateBlockUser, **authenticateUserToken**, **validateAdmin**, blockUser)